File: frontend/src/services/resourceService.js
import axiosInstance from "../utils/axiosInstance";

const API_URL = "/resources";

// Get all resources
const getAllResources = async () => {
    const response = await axiosInstance.get(API_URL);
    return response.data;
};

// Create resource
const createResource = async (resource) => {
    const response = await axiosInstance.post(API_URL, resource);
    return response.data;
};

// Update resource
const updateResource = async (id, resource) => {
    const response = await axiosInstance.put(`${API_URL}/${id}`, resource);
    return response.data;
};

// Delete resource
const deleteResource = async (id) => {
    const response = await axiosInstance.delete(`${API_URL}/${id}`);
    return response.data;
};

export default {
    getAllResources,
    createResource,
    updateResource,
    deleteResource,
};

File: backend/src/main/java/com/labresource/backend/service/ResourceService.java
package com.labresource.backend.service;

import com.labresource.backend.entity.Resource;
import com.labresource.backend.repository.ResourceRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ResourceService {

    private final ResourceRepository resourceRepository;

    public ResourceService(ResourceRepository resourceRepository) {
        this.resourceRepository = resourceRepository;
    }

    // Get all resources
    public List<Resource> getAllResources() {
        return resourceRepository.findAll();
    }

    // Get resource by ID
    public Resource getResourceById(Long id) {
        return resourceRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Resource not found"));
    }

    // Save resource
    public Resource saveResource(Resource resource) {
        return resourceRepository.save(resource);
    }

    // Update resource
    public Resource updateResource(Long id, Resource resource) {

        Resource existingResource = getResourceById(id);

        existingResource.setResourceName(resource.getResourceName());
        existingResource.setResourceType(resource.getResourceType());
        existingResource.setDescription(resource.getDescription());
        existingResource.setQuantity(resource.getQuantity());
        existingResource.setStatus(resource.getStatus());

        return resourceRepository.save(existingResource);
    }

    // Delete resource
    public void deleteResource(Long id) {

        Resource existingResource = getResourceById(id);

        resourceRepository.delete(existingResource);
    }
}